lib/Basics/CrashHandler.cpp

```diff
@@ -470,6 +474,60 @@ void crashHandlerSignalHandler(int signal, siginfo_t* info, void* ucontext) {
 }
 #endif
 
+#ifdef _WIN32
+
+static std::string miniDumpDirectory = "C:\\temp";
+
+// This function is called once after installing the exceptionFilter.
+void setMiniDumpDirectory() { miniDumpDirectory = TRI_GetTempPath(); }
+
+void createMiniDump(EXCEPTION_POINTERS* pointers) {
+  time_t rawtime;
+  time(&rawtime);
+  struct tm timeinfo;
+  _gmtime64_s(&timeinfo, &rawtime);
+  char time[20];
+  strftime(time, sizeof(time), "%Y-%m-%dT%H-%M-%S", &timeinfo);
+  
+  char filename[MAX_PATH];
+  sprintf(filename, "%s\\%s_%d_%d.dmp", miniDumpDirectory.c_str(), time, GetCurrentProcessId(), GetCurrentThreadId());
+  HANDLE hFile = CreateFile(filename, GENERIC_WRITE, 0, nullptr,
+                            CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+  if (hFile == INVALID_HANDLE_VALUE) {
+    LOG_TOPIC("ba80e", FATAL, arangodb::Logger::CRASH)
+        << "Could not open minidump file: " << GetLastError();
+    return;
+  }
+
+  MINIDUMP_EXCEPTION_INFORMATION exceptionInfo;
+  exceptionInfo.ThreadId = GetCurrentThreadId();
+  exceptionInfo.ExceptionPointers = pointers;
+  exceptionInfo.ClientPointers = FALSE;
+
+  if (MiniDumpWriteDump(GetCurrentProcess(), GetCurrentProcessId(), hFile,
```

Comment:
MSDN recommends to spawn an additional process for running minidump [see the Remarks](https://docs.microsoft.com/en-gb/windows/win32/api/minidumpapiset/nf-minidumpapiset-minidumpwritedump?redirectedfrom=MSDN). Or at least sync the calls as MiniDumpWriteDump is single threaded. Yeah it is very unlikely that two threads will have unhandled exception, but  nonetheless.

Reply:
I know, but I wanted to avoid the complexity of having an additional process. But you are right, we should at least sync the calls.